README.md

```diff
@@ -458,18 +458,45 @@ status:
+
+A `MappingContainer` object **MUST** define a `path` entry is a string containing a [JSONPath][jsonpath] that references container like locations in the target resource. The following expressions **MUST** be applied to each object matched by the path.  A `MappingTemplate` object **MAY** define `name`, as a string containing a [Restricted JSONPath](#restricted-jsonpath) that describes the location of a string in the target resource that names the container. A `MappingTemplate` object **MAY** define `env`, as a string containing a [Restricted JSONPath](#restricted-jsonpath) that describes the location of [`[]EnvVar`][envvar] array in the target resource. If not specified, the default `env` expression **MUST** be appropriate for mapping within an actual Container object (`.env`). A `MappingTemplate` object **MAY** define `volumeMounts`, as a string containing a [Restricted JSONPath](#restricted-jsonpath) that describes the location of [`[]VolumeMount`][volumemount] array in the target resource. If not specified, the default `env` expression **MUST** be appropriate for mapping within an actual Container object (`.volumeMounts`).
 
 [cwrm-crd]: servicebinding.io_clusterworkloadresourcemappings.yaml
 [container]: https://kubernetes.io/docs/reference/kubernetes-api/workloads-resources/container/
 [crd-syntax]: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition
 [envvar]: https://kubernetes.io/docs/reference/kubernetes-api/workloads-resources/container/#environment-variables
-[jsonpath]: https://kubernetes.io/docs/reference/kubectl/jsonpath/
+[jsonpath]: http://goessner.net/articles/JsonPath/
 [volume]: https://kubernetes.io/docs/reference/kubernetes-api/config-and-storage-resources/volume
 [volumemount]: https://kubernetes.io/docs/reference/kubernetes-api/workloads-resources/container/#volumes
 
+## Restricted JSONPath
+
+> Note: only expression labels as 'Restricted JSONPath' **MUST** conform to this requirement. Other expressions **MAY** use the full JSONPath syntax.
+
+A Restricted JSONPath is a subset of [JSONPath][jsonpath] expressions that **MUST NOT** use type and operators other than fields separated by the child operator.
```

Comment:
I agree for a user guide but not for the spec. We're not defining a floor of functionality in this case we're defining the ceiling.

Reply:
well, but json path is already defined as it is and there is no movement on that front. Hence, it is clearer to state here what subset of the the jsonpath spec we understand as "restricted jsonpath", Otherwise, we are forcing the developers to figure that out by themselves.

Reply:
I think the current text says what your asking for. Would you like to propose alternate language?

Reply:
Yes, I would like to propose that we say something like this:


A Restricted JSONPath is a subset of [JSONPath][jsonpath] expressions that **MUST** consists of dot-selectors only.

```
restricted-json-path = +(dot-selector)
dot-selector    = "." dot-member-name
   dot-member-name = name-first *name-char
   name-first =
                         ALPHA /
                         "_"   /           ; _
                         %x80-10FFFF       ; any non-ASCII Unicode character
   name-char = DIGIT / name-first

   DIGIT           =  %x30-39              ; 0-9
   ALPHA           =  %x41-5A / %x61-7A    ; A-Z / a-z   
```

IMHO, this describe exactly what is allowed by the spec.

Reply:
"selectors" are introduced in the rfc, it's strange to use them for our definition which isn't based on the rfc. Moreover, the dot selector does not fully cover the "child operator" which allows for field access via dot or square bracket notation.

I'm going to keep the current text instead of your proposal. The "child operator" is defined and the examples below this line clearly show what is allowed. We can revisit this once the rfc is no longer a draft and we switch all JSONPaths over to the rfc's definition.